staging/src/k8s.io/apiserver/pkg/admission/initializer/initializer.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type pluginInitializer struct {
+	externalClient    kubernetes.Interface
+	externalInformers informers.SharedInformerFactory
+	authorizer        authorizer.Authorizer
+}
+
+// New creates an instance of admission plugins initializer.
+// client config is optional by default in-cluster config is used to populate external clients and informers.
+func New(cfg *rest.Config, authz authorizer.Authorizer) (pluginInitializer, error) {
+	if cfg == nil {
+		var err error
+		cfg, err = rest.InClusterConfig()
+		if err != nil {
+			return pluginInitializer{}, err
+		}
+	}
+
+	extClientset, err := kubernetes.NewForConfig(cfg)
```

Comment:
Accept the clientset and informer factory as arguments.

The informer factory should be shared.  We only want a single instance or we'll end up with multiple list/watches running.

The clientset will be privileged and settings like QPS and roundtrippers should be configured externally and passed here.

I like the idea of including a loopback `*rest.Config` through the initializer so that people aren't forced to create their own initializers if they don't want to.